tchannel-core/src/main/java/com/uber/tchannel/handlers/RequestRouter.java

```diff
@@ -251,11 +251,14 @@ public void onFailure(Throwable e) {
             private void doRequestEndProcessing() {
                 if (span != null) {
                     span.finish();
+                    TracingContext tracingContext = topChannel.getTracingContext();
```

Comment:
imo any use of tracing context at this point will cause errors, because there is no guarantee that this Future callback will run on the same thread as line 227 that pushed the span to the context. That is why I suggest instead to add a call to `topChannel.getTracingContext().clear()` just before line 227, and in the callback only call `span.finish()`, without messing around with the context.